### PR TITLE
HPCC-21519 THOR - Make sure audience and message class is sensible

### PR DESCRIPTION
**Pull request by shamser**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by shamser** [comment]
@jakesmith Please can you review the logging changes in the Thor

**Comment by jakesmith** [comment]
@shamser - I'll review the 2nd commit, but is it intentional that this 2nd commit contains non-thor related changes too? e.g. dllserver / environment / roxiehelper ?

**Comment by shamser** [comment]
@jakesmith It shouldn't contain changes related to other components.  Leave the commit until I've looked into this.

**Comment by jakesmith** [comment]
@shamser - I've been holding off reviewing this, because it still contains a lot of non related files in the commit (+ the build failure).

**Comment by shamser** [comment]
@jakesmith Unrelated changes removed and rebased.  Please can you review?

File: thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp [hunk 1]
```diff
@@ -979,7 +979,7 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor
                         bool removed;
                         msg.read(removed);
                         if (!removed)
-                            WARNLOG("KJ service failed to remove [%u]", closeCmd);
+                            DBGLOG("KJ service failed to remove [%u]", closeCmd);
```
Comment (jakesmith) [inline thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp]: this should prob. be a IWARNLOG

File: thorlcr/graph/thgraph.cpp [hunk 2]
```diff
@@ -2724,9 +2724,13 @@ void CJobBase::startJob()
     {
         StringBuffer output;
         if (getAllStacks(output))
-            PrintLogDirect(output);
+        {
+#ifndef DISABLE_PRINTLOG
+            IERRLOG("%s", output.str());
+#endif
```
Comment (jakesmith) [inline thorlcr/graph/thgraph.cpp]: can remove the #ifndef - IERRLOG on it's own would be good.

File: thorlcr/graph/thgraph.cpp [hunk 3]
```diff
@@ -2724,9 +2724,13 @@ void CJobBase::startJob()
...
+        }
         else
-            WARNLOG("Failed to capture process stacks: %s", output.str());
+            DBGLOG("Failed to capture process stacks: %s", output.str());
```
Comment (jakesmith) [inline thorlcr/graph/thgraph.cpp]: IWARNLOG

File: thorlcr/graph/thgraphmaster.cpp [hunk 4]
```diff
@@ -692,7 +692,7 @@ class CBarrierMaster : public CInterface, implements IBarrier
                 e.setown(deserializeException(msg));
             sender = sender - 1; // 0 = master
             if (raisedSet->testSet(sender, true) && !aborted)
-                WARNLOG("CBarrierMaster, raise barrier message on tag %d, already received from slave %d", tag, sender);
+                DBGLOG("CBarrierMaster, raise barrier message on tag %d, already received from slave %d", tag, sender);
```
Comment (jakesmith) [inline thorlcr/graph/thgraphmaster.cpp]: IWARNLOG

File: thorlcr/graph/thgraphmaster.cpp [hunk 5]
```diff
@@ -1156,12 +1156,12 @@ class CThorCodeContextMaster : public CThorCodeContextBase
         {
             StringBuffer m;
             E->errorMessage(m);
-            DBGLOG("Unable to record exception in workunit: %s", m.str());
+            OERRLOG("Unable to record exception in workunit: %s", m.str());
```
Comment (jakesmith) [inline thorlcr/graph/thgraphmaster.cpp]: IWARNLOG

File: thorlcr/graph/thgraphmaster.cpp [hunk 6]
```diff
@@ -1156,12 +1156,12 @@ class CThorCodeContextMaster : public CThorCodeContextBase
...
             E->Release();
         }
         catch (...)
         {
-            DBGLOG("Unable to record exception in workunit: unknown exception");
+            OERRLOG("Unable to record exception in workunit: unknown exception");
```
Comment (jakesmith) [inline thorlcr/graph/thgraphmaster.cpp]: IWARNLOG

File: thorlcr/graph/thgraphmaster.cpp [hunk 7]
```diff
@@ -1177,12 +1177,12 @@ class CThorCodeContextMaster : public CThorCodeContextBase
         {
             StringBuffer m;
             E->errorMessage(m);
-            DBGLOG("Unable to record exception in workunit: %s", m.str());
+            OERRLOG("Unable to record exception in workunit: %s", m.str());
```
Comment (jakesmith) [inline thorlcr/graph/thgraphmaster.cpp]: IWARNLOG

File: thorlcr/graph/thgraphmaster.cpp [hunk 8]
```diff
@@ -1177,12 +1177,12 @@ class CThorCodeContextMaster : public CThorCodeContextBase
...
             E->Release();
         }
         catch (...)
         {
-            DBGLOG("Unable to record exception in workunit: unknown exception");
+            OERRLOG("Unable to record exception in workunit: unknown exception");
```
Comment (jakesmith) [inline thorlcr/graph/thgraphmaster.cpp]: IWARNLOG

File: thorlcr/master/thgraphmanager.cpp [hunk 9]
```diff
@@ -811,7 +811,7 @@ void CJobManager::reply(IConstWorkUnit *workunit, const char *wuid, IException *
         replyMb.append((unsigned)DAMP_THOR_REPLY_GOOD);
     if (!conversation->send(replyMb)) {
         s.clear();
-        ERRLOG("Failed to reply to agent %s",agentep.getUrlStr(s).str());
+        OERRLOG("Failed to reply to agent %s",agentep.getUrlStr(s).str());
```
Comment (jakesmith) [inline thorlcr/master/thgraphmanager.cpp]: IERRLOG

File: thorlcr/slave/slavmain.cpp [hunk 10]
```diff
@@ -2124,7 +2124,7 @@ class CFileInProgressHandler : public CSimpleInterface, implements IFileInProgre
         bakName.append((unsigned)dt.getSimple()).append("_").append((unsigned)GetCurrentProcessId()).append(".bak");
         iFileIO.clear(); // close old for rename
         iFile->rename(bakName.str());
-        WARNLOG("Renamed to %s", bakName.str());
+        OWARNLOG("Renamed to %s", bakName.str());
```
Comment (jakesmith) [inline thorlcr/slave/slavmain.cpp]: not sure why that was a warning. Change to DBGLOG

File: thorlcr/thorutil/thmem.cpp [hunk 11]
```diff
@@ -1236,7 +1236,7 @@ rowidx_t CThorExpandingRowArray::serializeBlock(MemoryBuffer &mb, rowidx_t idx,
             serializer->serialize(out,(const byte *)row);
         else if (warnnull)
         {
-            WARNLOG("CThorExpandingRowArray::serialize ignoring NULL row");
+            DBGLOG("CThorExpandingRowArray::serialize ignoring NULL row");
```
Comment (jakesmith) [inline thorlcr/thorutil/thmem.cpp]: IWARNLOG

File: thorlcr/thorutil/thormisc.cpp [hunk 12]
```diff
@@ -561,7 +561,7 @@ void SetLogName(const char *prefix, const char *logdir, const char *thorname, bo
     {
         if (!recursiveCreateDirectory(logdir))
         {
-            PrintLog("Failed to use %s as log directory, using current working directory", logdir); // default working directory should be open already
+            DBGLOG("Failed to use %s as log directory, using current working directory", logdir); // default working directory should be open already
```
Comment (jakesmith) [inline thorlcr/thorutil/thormisc.cpp]: OWARLOG

File: thorlcr/thorutil/thormisc.cpp [hunk 13]
```diff
@@ -594,8 +594,8 @@ void SetLogName(const char *prefix, const char *logdir, const char *thorname, bo
     logname.append(".log");
     StringBuffer lf;
     openLogFile(lf, logname.str());
-    PrintLog("Opened log file %s", lf.str());
-    PrintLog("Build %s", BUILD_TAG);
+    DBGLOG("Opened log file %s", lf.str());
+    DBGLOG("Build %s", BUILD_TAG);
```
Comment (jakesmith) [inline thorlcr/thorutil/thormisc.cpp]: PROGLOG

File: thorlcr/thorutil/thormisc.cpp [hunk 14]
```diff
@@ -594,8 +594,8 @@ void SetLogName(const char *prefix, const char *logdir, const char *thorname, bo
     logname.append(".log");
     StringBuffer lf;
     openLogFile(lf, logname.str());
-    PrintLog("Opened log file %s", lf.str());
-    PrintLog("Build %s", BUILD_TAG);
+    DBGLOG("Opened log file %s", lf.str());
```
Comment (jakesmith) [inline thorlcr/thorutil/thormisc.cpp]: PROGLOG

**Review by jakesmith — CHANGES_REQUESTED** [review]

**Comment by jakesmith** [comment]
@shamser - added some comments/suggestions

**Comment by shamser** [comment]
@jakesmith Modified as suggested.  Shall I squash?

**Comment by jakesmith** [comment]
@shamser - yes go ahead please.

**Comment by jakesmith** [comment]
@AttilaVamos - not clear from above - can you see which unittest failed?

**Comment by AttilaVamos** [comment]
@jakesmith It was the DFUAccessTests

**Comment by AttilaVamos** [comment]
@jakesmith It was the DFUAccessTests

**Comment by AttilaVamos** [comment]
Recheduled.

**Comment by jakesmith** [comment]
>It was the DFUAccessTests

@AttilaVamos - ok interesting, can you send me the logs / core ?

**Comment by AttilaVamos** [comment]
@jakesmith  sent via Flowdock

**Comment by jakesmith** [comment]
@shamser - please squash

**Comment by shamser** [comment]
@jakesmith Squashed.

**Comment by jakesmith** [comment]
@shamser - need re-targeting to master (for 7.4)

**Comment by richardkchapman** [comment]
I think some of the other changes in this series have gone in 7.2 - not sure whether it needs to be master?

**Comment by jakesmith** [comment]
ok, I wasn't sure. I thought as not a regression/fix - it shouldn't be a point build change.

**Comment by richardkchapman** [comment]
@shamser Are you ok with this going in 7.2.x?

**Comment by shamser** [comment]
@richardkchapman May be merge master so that we can get the other logging related changes in together.  What do you think?

**Comment by richardkchapman** [comment]
@shamser Fine by me

**Pull request merged**